source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -481,6 +486,109 @@ ShaderGraphPtr ShaderGraph::create(const ShaderGraph* parent, const string& name
         // Start traversal from this output
         root = output;
     }
+    else if (element->isA<Node>())
+    {
+        NodePtr node = element->asA<Node>();
+
+        NodeDefPtr nodeDef = node->getNodeDef();
+        if (!nodeDef)
+        {
+            throw ExceptionShaderGenError("Could not find a nodedef for node '" + node->getName() + "'");
+        }
+
+        graph = std::make_shared<ShaderGraph>(parent, name, element->getDocument(), context.getReservedWords());
+
+        // Create input sockets
+        graph->addInputSockets(*nodeDef, context);
+
+        // Create output sockets
+        graph->addOutputSockets(*nodeDef);
+
+        // Create this shader node in the graph.
+        const string& newNodeName = node->getName();
+        ShaderNodePtr newNode = ShaderNode::create(graph.get(), newNodeName, *nodeDef, context);
+        graph->_nodeMap[newNodeName] = newNode;
+        graph->_nodeOrder.push_back(newNode.get());
+
+        // Connect it to the graph output
+        //
+        // TODO: Support multioutput nodes
+        //
+        ShaderGraphOutputSocket* outputSocket = graph->getOutputSocket();
```

Comment:
Can not all the output sockets information be updated even if not evaluated?

Reply:
Will try that out.